ambari-server/src/main/resources/common-services/AMBARI_INFRA_SOLR/0.1.0/package/scripts/setup_infra_solr.py

```diff
@@ -91,13 +91,13 @@ def setup_infra_solr(name = None):
            owner=params.infra_solr_user,
            group=params.user_group,
            mode=0640)
-
-    File(os.path.join(params.limits_conf_dir, 'infra-solr.conf'),
-         owner='root',
-         group='root',
-         mode=0644,
-         content=Template("infra-solr.conf.j2")
-         )
+    if os.path.exists(params.limits_conf_dir):
+      File(os.path.join(params.limits_conf_dir, 'infra-solr.conf'),
```

Comment:
I would have guessed this needs a corresponding change in the unit test:

https://github.com/apache/ambari/blob/11ea165b7c6d8da5462011fff10b6420b4f410c1/ambari-server/src/test/python/stacks/2.4/AMBARI_INFRA/test_infra_solr.py#L108-L112

but it was disabled in 2160d7a0d02 temporarily:

https://github.com/apache/ambari/blob/11ea165b7c6d8da5462011fff10b6420b4f410c1/ambari-server/src/test/python/stacks/2.4/AMBARI_INFRA/test_infra_solr.py#L132-L133